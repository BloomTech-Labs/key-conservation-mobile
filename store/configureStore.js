import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

let composeEnhancers = compose;

if (__DEV__) {
  composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const store = createStore(rootReducer, composeEnhancers(applyMiddleware(thunk)));

export default store;