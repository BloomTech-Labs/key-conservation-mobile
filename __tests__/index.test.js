import configureStore from 'redux-mock-store';
const {
  setCampaign,
  toggleCampaignText,
  getCampaignPost,
  addBookmark,
  removeBookmark,
  getConnections,
} = require('../store/actions');
import thunk from 'redux-thunk';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

test('setCampaign("test")', () => {
  // Initialize mockstore with empty state
  const initialState = {};
  const store = mockStore(initialState);

  const fakeCampaign = { test: 'test' };

  store.dispatch(setCampaign(fakeCampaign));

  expect(store.getActions()).toMatchSnapshot();
});

test('setCampaign(null)', () => {
  // Initialize mockstore with empty state
  const initialState = {};
  const store = mockStore(initialState);

  store.dispatch(setCampaign());

  expect(store.getActions()).toMatchSnapshot();
});

test('toggleCampaignText', () => {
  const initialState = {};
  const store = mockStore(initialState);
  store.dispatch(toggleCampaignText());
  expect(store.getActions()).toMatchSnapshot();
});

describe('getCampaignPost', () => {
  const fakeCampaignPost = { test: 'test', id: 1 };
  let { id, test } = fakeCampaignPost;
  const initialState = {};
  const store = mockStore(initialState);

  const getCampaigns = () => {
    return (dispatch) => {
      getCampaignPost({ id });

      return dispatch({ type: 'GET_REPORTS_SUCCESS' });
    };
  };

  it('should fetch proper post', () => {
    store.dispatch(getCampaigns());

    const actions = store.getActions();
    expect(actions[0]).toEqual({ type: 'GET_REPORTS_SUCCESS' });
  });
});

describe('addBookmark', () => {
  const fakeBookmark = { test: 'test', id: 1 };
  let { id, test } = fakeBookmark;
  const initialState = {};
  const store = mockStore(initialState);

  const bookmark = () => {
    return (dispatch) => {
      addBookmark({ id });

      return dispatch({ type: 'ADD_BOOKMARK_SUCCESS' });
    };
  };

  it('should add bookmark', () => {
    store.dispatch(bookmark());

    const actions = store.getActions();
    expect(actions[0]).toEqual({ type: 'ADD_BOOKMARK_SUCCESS' });
  });
});

describe('removeBookmark', () => {
  const fakeBookmark = { test: 'test', id: 1 };
  let { id, test } = fakeBookmark;
  const initialState = {};
  const store = mockStore(initialState);

  const bookmarkRemove = () => {
    return (dispatch) => {
      removeBookmark({ id });

      return dispatch({ type: 'REMOVE_BOOKMARK_SUCCESS' });
    };
  };

  it('should remove bookmark', () => {
    store.dispatch(bookmarkRemove());

    const actions = store.getActions();
    expect(actions[0]).toEqual({ type: 'REMOVE_BOOKMARK_SUCCESS' });
  });
});

test('getConnections', () => {
  const initialState = {};
  const store = mockStore(initialState);

  store.dispatch(getConnections());
  expect(store.getActions()).toMatchSnapshot();
});
