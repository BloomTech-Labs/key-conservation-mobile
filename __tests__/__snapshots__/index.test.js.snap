// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getConnections 1`] = `Array []`;

exports[`setCampaign("test") 1`] = `
Array [
  Object {
    "payload": Object {
      "test": "test",
    },
    "type": "SET_CAMPAIGN",
  },
]
`;

exports[`setCampaign(null) 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "SET_CAMPAIGN",
  },
]
`;

exports[`toggleCampaignText 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "TOGGLE_CAMPAIGN_TEXT",
  },
]
`;
